openapi-bal-service/src/main/java/io/ballerina/openapi/converter/service/OpenAPIRequestBodyMapper.java

```diff
@@ -234,19 +357,36 @@ private void handleArrayTypePayload(Map<String, Schema> schema, ArrayTypeDescrip
             Schema itemSchema = new Schema();
             arraySchema.setItems(itemSchema.$ref(ConverterCommonUtils.unescapeIdentifier(
                     referenceNode.name().text().trim())));
-            io.swagger.v3.oas.models.media.MediaType media = new io.swagger.v3.oas.models.media.MediaType();
             media.setSchema(arraySchema);
-            if (requestBody.getContent() != null) {
-                Content content = requestBody.getContent();
-                content.addMediaType(mimeType, media);
+        } else if (typeDescriptorNode.kind() == SyntaxKind.BYTE_TYPE_DESC) {
+            StringSchema byteSchema = new StringSchema();
+            byteSchema.setFormat("byte");
+            media.setSchema(byteSchema);
+        } else {
+            Schema itemSchema = getOpenApiSchema(typeDescriptorNode.kind());
+            arraySchema.setItems(itemSchema);
+            media.setSchema(arraySchema);
+        }
+
+        if (requestBody.getContent() != null) {
+            Content content = requestBody.getContent();
+            if (content.containsKey(mimeType)) {
+                ComposedSchema oneOfSchema = new ComposedSchema();
+                oneOfSchema.addOneOfItem(content.get(mimeType).getSchema());
+                oneOfSchema.addOneOfItem(arraySchema);
+                media.setSchema(oneOfSchema);
             } else {
-                requestBody.setContent(new Content().addMediaType(mimeType, media));
-            }
-            //  Adding conditional check for http delete operation as it cannot have body
-            //  parameter.
-            if (!operationAdaptor.getHttpOperation().equalsIgnoreCase("delete")) {
-                operationAdaptor.getOperation().setRequestBody(requestBody);
+                content.addMediaType(mimeType, media);
             }
+            content.addMediaType(mimeType, media);
+            requestBody.setContent(content);
+        } else {
+            requestBody.setContent(new Content().addMediaType(mimeType, media));
+        }
+        //  Adding conditional check for http delete operation as it cannot have body
+        //  parameter.
+        if (!operationAdaptor.getHttpOperation().equalsIgnoreCase("delete")) {
```

Comment:
Do we have any constants already defined for "delete"? If so we can use it

Reply:
Fixed here: https://github.com/ballerina-platform/openapi-tools/pull/1544/commits/213d3e3838ce199712f5d0a24d321617db6d99c2